optuna/_transform.py

```diff
@@ -90,6 +90,12 @@ def column_to_encoded_columns(self) -> List[numpy.ndarray]:
     def encoded_column_to_column(self) -> numpy.ndarray:
         return self._encoded_column_to_column
 
+    @property
+    def num_params(self) -> int:
+        return len(self._column_to_encoded_columns)
```

Comment:
I have three suggestions.

1. [nits] `n_params` is more consistent with other variable names in Optuna (e.g. `n_trials` and `n_trees`).
2. `SearchSpaceTransform` transforms the search space between Optuna's search space representation and numpy-based representation. It is difficult to know `num_params` means which ones. `n_original_params` or `n_optuna_params` might be more clear. What do you think?
3. Not a strong opinion but `len(self._search_space)` is more intuitive I think.

```suggestion
    @property
    def n_original_params(self) -> int:
        return len(self._search_space)
```

Also could you update the Attributes section of docstring.

Reply:
I thought the domain of `_SearchSpaceTransform` is called "parameters" (as they are the same as the user-specified parameters) and the range is called "features" (as in "one-hot features"). I guess adding a `n_features` property alongside with `n_params` would make it clear.

Reply:
Another possible solution: make `self._search_space` public and remove `num_params` property.
What do you think?

Reply:
> I thought the domain of _SearchSpaceTransform is called "parameters" (as they are the same as the user-specified parameters) and the range is called "features" (as in "one-hot features").

I checked the docstring of `_SearchSpaceTransform` but I cannot find the term `features`. For example, the description of `column_to_encoded_columns` is `Constructed mapping from original parameter column index to encoded column indices.`. What do you think?

https://github.com/optuna/optuna/blob/93e23d6a7cd5ed126d669427884e409d15292196/optuna/_transform.py#L18-L57

Reply:
> Another possible solution: make self._search_space public and remove num_params property (which will be no more than len(trans.search_space)). What do you think?

We don't need to make `search_space` public since it is just referred at `n_original_params` (or `num_params`) property, that is defined in `_SearchSpaceTransform` class.

Reply:
No, I mean instead of making a new `num_params` property, we can directly use `len(trans.search_space)` from `ImportanceEvaluator`s.

Reply:
> No, I mean instead of making a new num_params property, we can directly use len(trans.search_space) from ImportanceEvaluators.

Ah, that sounds make sense. I think we still don't need to make it public because `search_space` is a constructor argument.